project/FreeGen.scala

```diff
@@ -152,17 +154,17 @@ class FreeGen(managed: List[Class[_]], log: Logger) {
   // Ctor values for all methods in of A plus superclasses, interfaces, etc.
   def ctors[A](implicit ev: ClassTag[A]): List[Ctor] =
     methods(ev.runtimeClass).groupBy(_.getName).toList.flatMap { case (n, ms) =>
-      ms.toList.zipWithIndex.map {
+      ms.toList.sortBy(_.getGenericParameterTypes.map(toScalaType).mkString(",")).zipWithIndex.map {
```

Comment:
Thanks for this change, I should have done this a long time ago. Stuff was getting shuffled around every time I ran freegen.